### ISSUE TITLE
feat(reductions): masked sample statistics — `masked_mean` / `masked_std` over a leading axis with a variance floor

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#241]
## Summary

Add mask-aware mean / standard-deviation reductions over a leading sample axis (time or ensemble) so that per-gridpoint standardisation of fields on masked domains is correct: land cells must not contribute to statistics and must not produce zero-variance divisions.

**Motivated by**: jejjohnson/somax#170 — `StateAffine.from_samples(states, per_gridpoint=True, mask=...)` (jejjohnson/somax#171) computes `loc = mean₀(x)`, `scale = max(std₀(x), ε)` per field. On masked grids this needs the same `jnp.where(mask.h, ..., 0)` hygiene that `area_mean` already applies, plus a floor on the std. The helpers belong in finitevolx next to the existing reductions so both somax and standalone users get one tested implementation.

## What to build

**File**: `finitevolx/_src/operators/reductions.py`

```python
def masked_mean(
    samples: Float[Array, "N ..."],
    mask: Mask2D | Mask3D | None = None,
    *,
    location: Literal["h", "u", "v", "xy_corner"] = "h",
    axis: int | tuple[int, ...] = 0,
) -> Float[Array, "..."]:
    """Mean over `axis`; dry cells (per `location` mask) return 0."""

def masked_std(
    samples: Float[Array, "N ..."],
    mask: Mask2D | Mask3D | None = None,
    *,
    location: Literal["h", "u", "v", "xy_corner"] = "h",
    axis: int | tuple[int, ...] = 0,
    eps: float = 1e-8,
    ddof: int = 0,
) -> Float[Array, "..."]:
    """Std over `axis`, floored at `eps`; dry cells return 1.0 (identity scale)."""

def masked_moments(samples, mask=None, *, location="h", axis=0, eps=1e-8) -> tuple[Array, Array]:
    """(mean, std) in one pass (Welford or two-pass with masking)."""
```

- `axis=0` gives per-gridpoint statistics (shape `[Ny, Nx]` or `[Nz, Ny, Nx]`); `axis=(0, -2, -1)` gives per-field scalars (with masks, only wet cells enter the reduction — reuse `_apply_mask_2d` / `_apply_mask_3d` and divide by the wet count).
- `location` picks `mask.h` / `mask.u` / `mask.v` / `mask.xy_corner` so U- and V-point fields are handled on the C-grid.
- Return values on dry cells are `(0, 1)` so that `(x − mean) / std` leaves land at zero and the inverse map is well defined.
- NaN / Inf sentinels on land must not contaminate wet-cell statistics (same guarantee as `area_sum` / `area_mean`).

## Public API additions

```python
"masked_mean", "masked_std", "masked_moments",
```

## Testing

- No mask: matches `jnp.mean` / `jnp.std` (with `ddof`) exactly
- Masked: wet-cell statistics match a NumPy reference computed on the wet subset; dry cells return `(0, 1)`
- NaN on land does not leak into wet cells
- Constant field: `std == eps`, not 0
- `location="u"` / `"v"` use the corresponding C-grid masks
- Per-field scalar reduction (`axis=(0,-2,-1)`) divides by the wet count, not `Ny·Nx`
- JIT + grad compatibility (grad of `masked_mean` w.r.t. `samples` is `mask / N`)

## Relationships

- Parent: jejjohnson/somax#170 (non-dimensionalisation & standardisation theme)
- Related: jejjohnson/somax#171 (`StateAffine.from_samples` will delegate here), #185 (realistic masks — where this matters most), `area_mean` / `volume_mean` in `reductions.py`
